pinot-common/src/main/java/org/apache/pinot/common/metrics/AbstractMetrics.java

```diff
@@ -707,15 +708,16 @@ public void removeTableGauge(final String tableName, final G gauge) {
 
 
   /**
+   * @deprecated please use removeTableGauge(final String tableName, final String key, final G gauge) instead
+   *
    * Removes a table gauge given the table name and the gauge.
    * The add/remove is expected to work correctly in case of being invoked across multiple threads.
    * @param tableName table name
+   * @param partitionId The partition id
    * @param gauge the gauge to be removed
    */
   public void removePartitionGauge(final String tableName, final int partitionId, final G gauge) {
```

Comment:
We also want to deprecate this because we don't want to add a separate method for each specific use case. We can use the general one `removeTableGauge(final String tableName, final String key, final G gauge)`

Reply:
then shall we add a new method for this and change all the existing reference to the new one?

Reply:
Yes, we have this new method in place. I will have several other PRs to consolidate different methods.

This PR is raise specifically to fix a comment to an already closed PR.

Reply:
I like to keep this method, since it provides a single way to add partition to a metric name. If we deprecate this in favor of string, then we will be left with what the developer decides at the time they code -- be it a dash, or undescore, or dot, or whatever.

@jugomezv  to comment since he added the method.

Reply:
That is correct @mcvsubbu, in fact it was @navina who suggested we add the per partition Gauge callback so that users dont have to be adding partition id on the name etc.

Reply:
@zhtaoxiang what is your proposed solution for ensuring that partition level metric names have a unified pattern? That was the motivation behind my comment in @jugomezv 's PR and it is not specific to any use-case. 
Does your PR here solve this problem or not?

Reply:
@mcvsubbu @jugomezv 

Since you have strong opinion on this, let's keep it. (Although we cannot stop users from using any string as metric name since `removeGauge(final String gaugeName)` is a publicly available method.)

Reply:
@navina that's a great question!

Personally, I feel that even with this method, we cannot 100% ensure that partition level metric names have a unified pattern (because users may abuse `setValueOfTableGauge(final String tableName, final G gauge, final long value)` to emit partition level metrics).

At the same time, I agree with you, @mcvsubbu and @jugomezv that it's highly possible that people choose to use this method if we have it.

So let's keep it.

Reply:
Sounds good @zhtaoxiang ! 
I think if this method exist, then it can potentially prevent the abuse of `setValueOfTableGauge` that you mentioned.

Reply:
Thanks a lot @zhtaoxiang I think it is best to encapsulate this partition and table related handling here to avoid each application from doing their own stuff, at least here we can try to set a standard for names when partition is involved. But its is true that devs can do whatever they want when building the metric name but hopefully reviewers will keep those names in check...

Reply:
also I think the addPartitionCallbackGauge is marked deprecated in the other PR you merged, can we unmark that as well, same rationals explained below

Reply:
@jugomezv I can do it. 

But can I change it to `setOrUpdatePartitionGauge`? The reason is that the method does not add a `callback`, instead, it needs a value `supplier`

Reply:
currently I see the code is as follows:
```
 public void addCallbackPartitionGaugeIfNeeded(final String tableName, final int partitionId, final G gauge,
      final Callable<Long> valueCallback) {
    final String fullGaugeName = composeTableGaugeName(tableName, String.valueOf(partitionId), gauge);
    addCallbackGaugeIfNeeded(fullGaugeName, valueCallback);
  }
```

so I dont see the Supplier being passed, I also see that the call this used is guarded to not add again so current name makes sense, may be I am missing something?:
public void addCallbackGaugeIfNeeded(final String metricName, final Callable<Long> valueCallback) {
```
    if (!_gaugeValues.containsKey(metricName)) {
      synchronized (_gaugeValues) {
        if (!_gaugeValues.containsKey(metricName)) {
          _gaugeValues.put(metricName, new AtomicLong(0L));
          addCallbackGauge(metricName, valueCallback);
        }
      }
    }
  }
```

Reply:
Let me break it down into details:
1. The last parameter of this method is not a `callback` but rather a callable, so I'd like to remove `callback` from the method name. `Callback` normally means it will tell the caller something.
2. The guard is actually not needed, `addCallbackGauge` itself will only add the gauge once, following calls cannot update it (I tested it and find this behavior, that's why I added updatable gauge in another PR https://github.com/apache/pinot/pull/9961). I have a plan to remove `_gaugeValues` entirely.
3. Can you please help me understand why we only need to add the gauge for the first time? I see only one usage, and it seems that it's okay to update the existing gauge.

Reply:
@jugomezv and I had a meeting, we agreed to
1/ rename the function `addCallbackPartitionGaugeIfNeeded` and revise the implementation
2/ run tests to confirm the change does not break anything

Reply:
I have updated the functions, and here is the metric emitted. 
<img width="877" alt="Screenshot 2023-01-12 at 5 03 05 PM" src="https://user-images.githubusercontent.com/9796617/212214164-26d79497-8020-4e2d-bc6a-61762f307789.png">